### PR TITLE
document the ~ notation

### LINKED ISSUE #236: Write about ~+~ operator forms in the tutorial
If e.g. `+` is a binary operator, one can write `~+~` to refer to it as itself (e.g. to pass it to a higher-order function, to ask for its type, etc.).  For example:
```
Disco> :type ~+~
~+~ : ℕ × ℕ → ℕ
Disco> import list
Loading list.disco...
Disco> foldr(~+~,0,[1 .. 10])
55
```
In general, one puts a `~` anywhere an argument is expected.  So e.g. `~+~ and ~*~ but also ~! (factorial) and -~ (negation)`. Currently I do not think this is mentioned in the tutorial.

### PR DESCRIPTION
**Pull request by TristanCacqueray**
Fixes #236

### CODE REVIEW

File: docs/tutorial/functions.rst [hunk 1]
```diff
@@ -167,3 +167,21 @@ not a multiplicative term.
...
+    Disco> :type ~+~
+    ~+~ : ℕ × ℕ → ℕ
+    Disco> import list
+    Loading list.disco...
+    Disco> foldr(~+~,0,[1 .. 10])
+    55
+
+.. note::
+
+    For unary operator, only one ``~`` is needed, e.g. ``~!`` for negation.
```
Comment (byorgey) [inline docs/tutorial/functions.rst:187]: `~!` is actually factorial, not negation.

Comment (TristanCacqueray) [reply docs/tutorial/functions.rst:187]: Oops :) So what is the logic for the position of the tild, should not negation be `~-` ?

Comment (byorgey) [reply docs/tutorial/functions.rst:187]: Negation is `-~`.  The tilde goes wherever the argument to the operator would go.  So postfix operators go after the tilde, and prefix operators go before the tilde.

**Comment by byorgey** [comment]
Thanks!

**Pull request merged**